src/resmom/linux/alps.c

```diff
@@ -4680,6 +5435,21 @@ alps_create_reserve_request(job *pjob, basil_request_reserve_t **req)
 		} else if (strcmp(BASIL_VAL_X2, arch) == 0) {
 			p->arch = basil_node_arch_x2;
 		}
+		if (pstate > 0) {
+			p->pstate = pstate;
+		}
+		if (pgov != NULL) {
+			if (strlen(pgov) < sizeof(p->pgovernor)) {
+				strcpy(p->pgovernor, pgov);
+			} else {
+				sprintf(log_buffer, "pgov value %s is too long,"
+					" length must be less than %ld",
+					pgov, sizeof(p->pgovernor));
+				log_event(PBSEVENT_DEBUG, PBS_EVENTCLASS_JOB,
+					LOG_DEBUG, pjob->ji_qs.ji_jobid,
+					log_buffer);
+			}
+		}
```

Comment:
This code seems outside the scope of using the new BASIL 1.3/1.4 interface to fill in information for resources PBS already has.